pkg/kubectl/apply/parse/openapi.go

```diff
@@ -150,11 +170,35 @@ type arraySchemaVisitor struct {
 func (v *arraySchemaVisitor) VisitArray(result *proto.Array) {
 	v.Result = result
 	v.Kind = "array"
+	v.Err = copyPatchStrategy(result.Path.String(), result.GetExtensions(), result.SubType.GetExtensions())
+}
+
+func copyPatchStrategy(field string, from, to map[string]interface{}) error {
+	if ext, found := from["x-kubernetes-patch-strategy"]; found {
+		strategy, ok := ext.(string)
+		if !ok {
+			return fmt.Errorf("Expected string value for x-kubernetes-patch-strategy on %s, was %T",
+				field, ext)
+		}
+		if strings.Contains(strategy, ",") {
```

Comment:
I don't understand what this block of code is doing?

Reply:
added comments